src/parser/mod.rs

```diff
@@ -10595,6 +10593,16 @@ impl<'a> Parser<'a> {
         self.parse_parenthesized_column_list_inner(optional, allow_empty, |p| p.parse_identifier())
     }
 
+    /// Parses a parenthesized comma-separated list of index columns, which can be arbitrary
+    /// expressions with ordering information (and an opclass in some dialects).
+    pub fn parse_parenthesized_index_column_list(
```

Comment:
```suggestion
    fn parse_parenthesized_index_column_list(
```
We can have this within the crate unless it would be required/useful to export?